infra/event-sync/src/main/java/com/simprints/infra/eventsync/sync/up/tasks/EventUpSyncTask.kt

```diff
@@ -170,45 +172,48 @@ internal class EventUpSyncTask @Inject constructor(
     ) = flow {
         Simber.tag(SYNC_LOG_TAG)
             .d("Uploading event scope - $eventScopeTypeToUpload in batches of $batchSize")
-        val sessionScopes = getClosedScopesForType(eventScopeTypeToUpload)
 
-        // Re-emitting the number of uploaded corrupted events
-        attemptInvalidEventUpload(
-            projectId,
-            sessionScopes.getCorruptedScopes()
-        ).collect { emit(it) }
+        while (eventRepository.getClosedEventScopesCount(eventScopeTypeToUpload) > 0 && currentCoroutineContext().isActive) {
```

Comment:
I'm not sure how much time it takes to count the closed event scopes, but I think if you use a `do while` you can avoid doing it by checking if the size of the `sessionScopes` is lower than the batch size

Reply:
That's how I did it initially but changed it due to concerns about error handling and creating an endless loop. @alexandr-simprints thoughts on performance vs robustness?

Reply:
> how much time it takes to count the closed event scopes
Insignificantly small amount of time, I can't see a need for an upfront optimization here

Reply:
Ok thanks for the explanation :)